proctor-common/src/main/java/com/indeed/proctor/common/ForceGroupsOptionsStrings.java

```diff
@@ -62,26 +71,105 @@ public static ForceGroupsOptions parseForceGroupsString(@Nullable final String f
             }
             //  bucketValueStart should now be the index of the minus sign or the first digit in a run of digits going to the end of the word
             final String testName = piece.substring(0, bucketValueStart).trim();
```

Comment:
Since this method is getting bigger, this might be a good time to extract some parts into a different method for readability. For example, we could define a method `Optional<Pair<String, Integer>> parseGroupString(final String piece)` that returns `testName` and `bucketValue` here.

Reply:
I thought pulling out the functionality of getting the bucket value would be a little cleaner so pulled that out. Can split if up more if desired.